app/api/__init__.py

```diff
@@ -82,35 +87,36 @@
           '/users-events-roles/<int:users_events_role_id>/role')
 
 # role_invites
-api.route(RoleInviteList, 'role_invite_list', '/events/<int:event_id>/role-invites',
+api.route(RoleInviteList, 'role_invite_list', '/role-invites', '/events/<int:event_id>/role-invites',
           '/events/<event_identifier>/role-invites')
 api.route(RoleInviteDetail, 'role_invite_detail', '/role-invites/<int:id>')
 api.route(RoleInviteRelationship, 'role_invite_event', '/role-invites/<int:id>/relationships/event')
 api.route(RoleInviteRelationship, 'role_invite_role', '/role-invites/<int:id>/relationships/role')
 
 # users_events_roles
-api.route(UsersEventsRolesList, 'users_events_role_list', '/events/<int:event_id>/users-events-roles',
+api.route(UsersEventsRolesList, 'users_events_role_list', '/users-events-roles',
+          '/events/<int:event_id>/users-events-roles',
           '/events/<event_identifier>/users-events-roles')
 api.route(UsersEventsRolesDetail, 'users_events_role_detail', '/users-events-roles/<int:id>')
 api.route(UsersEventsRolesRelationship, 'users_events_role_event', '/users-events-roles/<int:id>/relationships/event')
 api.route(UsersEventsRolesRelationship, 'users_events_role_role', '/users-events-roles/<int:id>/relationships/role')
 api.route(UsersEventsRolesRelationship, 'users_events_role_user', '/users-events-roles/<int:id>/relationships/user')
 
 # tickets
-api.route(TicketList, 'ticket_list', '/events/<int:event_id>/tickets',
+api.route(TicketList, 'ticket_list', '/tickets', '/events/<int:event_id>/tickets',
           '/events/<event_identifier>/tickets', '/ticket-tags/<int:ticket_tag_id>/tickets',
           '/access-codes/<int:access_code_id>/tickets')
 api.route(TicketDetail, 'ticket_detail', '/tickets/<int:id>', '/attendees/<int:attendee_id>/ticket')
-api.route(TicketRelationship, 'ticket_event', '/tickets/<int:id>/relationships/event')
-api.route(TicketRelationship, 'ticket_ticket_tag', '/tickets/<int:id>/relationships/ticket-tags')
-api.route(TicketRelationship, 'ticket_access_code', '/tickets/<int:id>/relationships/access-codes')
+api.route(TicketRelationshipRequired, 'ticket_event', '/tickets/<int:id>/relationships/event')
```

Comment:
Required means the relationship must be present in the request payload for a post request, and the relationship can be updated but it cannot be deleted. Also post on endpoints like `tickets/<int:id>/relationships/event` is disabled because it can be associated with one event. Optional relationship is just the usual relationship we had earlier, you can delete it, post it multiple times and it is not cumpolsary to provide it in the request payload